### PR TITLE
Customize Android biometric prompt behavior

### PR DESCRIPTION
**Pull request by patrickhammond**
There is currently no way to customize the Android biometric prompt. This PR addresses to customizations:
- Messaging on the biometric prompt (which was already half way in place).
- Allowing a biometric prompt to not require user confirmation.
- Updated the example app to show usage without breaking backwards compatability.

### CODE REVIEW

**Comment by hpoul** [comment]
hi,
thanks for your PR, just a few minor thoughts.

How about calling it something like `AndroidPromptInfo` (since you already found a config option which is not a message :-) ), so there is only one configuration object for each platform.. and i doubt that it's possible to unify  android and ios/mac (afaik on ios/mac there is only a single "message" to be displayed)

I also think that we could move the default message values from kotlin into the dart side, and make the kotlin side require those values to be set.. this should get rid of all those `if` statements in `toJson`? :-)

**Comment by patrickhammond** [comment]
@hpoul I think all of those seem like solid suggestions! Went back and forth on the Android specific configuration in the initial version but definitely feel more comfortable with it being called out as the platform configuration.

Updated the PR with those changes.

File: lib/src/biometric_storage.dart [hunk 1]
```diff
@@ -64,6 +64,49 @@ class StorageFileInitOptions {
...
+      json['negativeButton'] = negativeButton;
+    }
+
+    if (confirmationRequired != null) {
+      json['confirmationRequired'] = confirmationRequired;
+    }
+
+    return json;
+  }
+}
```
Comment (hpoul) [inline lib/src/biometric_storage.dart:108]:
```suggestion
  Map<String, dynamic> _toJson() => <String, dynamic>{
        'title': title,
        'subtitle': subtitle,
        'description': description,
        'negativeButton': negativeButton,
        'confirmationRequired': confirmationRequired,
      };
```

File: lib/src/biometric_storage.dart [hunk 2]
```diff
@@ -64,6 +64,49 @@ class StorageFileInitOptions {
...
 }
 
+class AndroidPromptInfo {
+  const AndroidPromptInfo({
+    this.title = 'Authenticate to unlock data',
+    this.subtitle,
+    this.description,
+    this.negativeButton = 'Cancel',
+    this.confirmationRequired = true,
+  });
```
Comment (hpoul) [inline lib/src/biometric_storage.dart:74]:
```suggestion
  })  : assert(title != null),
        assert(negativeButton != null),
        assert(confirmationRequired != null);
```

**Comment by toddtarbox** [comment]
I'm looking for this customization as well. Nice work! What's the ETA on getting this through? Just curious really.

**Comment by patrickhammond** [comment]
@hpoul I can't think of any reason to not include those changes. Added them in. Thanks!

**Comment by hpoul** [comment]
@toddtarbox is published as 0.3.4

**Comment by toddtarbox** [comment]
Thanks all! Nicely done.

**Pull request merged**